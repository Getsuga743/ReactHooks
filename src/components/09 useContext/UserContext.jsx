/* eslint-disable import/prefer-default-export */
import { createContext } from 'react';

const UserContext = createContext(null);

export default UserContext;
