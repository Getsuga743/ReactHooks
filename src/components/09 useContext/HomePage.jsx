import React, { useContext } from 'react';
import UserContext from './UserContext';

const HomePage = () => {
  // busque la siguiente istancia de UserContext
  // pasamos un objeto como el value del UserContext
  // lo desetructuramos, el objeto es el use y el setUser
  const { user } = useContext(UserContext);
  return (
    <div className="container-fluid">
      <h1>HomePage</h1>
      <h2>Hello</h2>
      {JSON.stringify(user, null, 3)}
    </div>
  );
};

export default HomePage;
