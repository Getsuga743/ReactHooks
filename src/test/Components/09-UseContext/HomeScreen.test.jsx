/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import HomePage from '../../../components/09 useContext/HomePage';
import UserContext from '../../../components/09 useContext/UserContext';
import '../../../setupTest';

describe('pruebas en <HomePage />', () => {
  const user = {
    name: 'Fernando',
    email: 'fernando@gmail.com',
  };
  // shallow hace una renderizacion unicamente del componente q le especifico,
  // home page nose muestra
  // para eso usamos mount
  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomePage />
    </UserContext.Provider>,
  );

  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
