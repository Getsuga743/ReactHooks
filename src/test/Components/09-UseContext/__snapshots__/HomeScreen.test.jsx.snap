// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <HomePage /> debe de mostrarse correctamente 1`] = `
<div
  className="container-fluid"
>
  <h1>
    HomePage
  </h1>
  <h2>
    Hello
  </h2>
  {
   "name": "Fernando",
   "email": "fernando@gmail.com"
}
</div>
`;
